swap_meet/vendor.py

```diff
@@ -1,2 +1,146 @@
+# simon? -- or whoever may be grading this project, please disregard
+# this commit I am working on the projects over the break to do 
+#corrections pls lmk if I should wait until after already graded that
+# I can go over them thank you!
+
 class Vendor:
-    pass
\ No newline at end of file
+    
+# Wave 01
+    def __init__(self, inventory = []):
+        self.inventory = inventory
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+
+        return False
+
+
+# Wave 02
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+
+
+
+# Wave 03 
+    def swap_items(self, other_vendor, my_item, their_item):
+        #
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            self.inventory.remove(my_item)
+            other_vendor.inventory.append(my_item)
+            other_vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+            return True
+        
+        return False
+
+
+# Wave 04
+    
+    def swap_first_item(self, other_vendor):
+        
+        if not self.inventory or not other_vendor.inventory:
+            return False
+
+        vendor_first_item = self.inventory[0]
+        other_vendor_first_item = other_vendor.inventory[0]
+
+        if not vendor_first_item or not other_vendor_first_item:
+            return False
```

Comment:
Realistically (aka if we were working together on industry), I probably wouldn't comment on this, because it just feels like explicit, redundant, readable code.

However, as a teacher, I'd like to point out: I think that you don't need this conditional, logically. Above on lines 50-51, you check if `self.inventory` and `other_vendor.inventory` are Falsy (`if not self.inventory ...`). This is the same as saying "If `self.inventory are empty" (aka are Falsy), then you'd `return False`. Thus, I believe that these two conditionals are the same. Let me know if you have questions on this!